smell-test

```diff
@@ -33,38 +35,78 @@ def valid_ip(address):
     except:
         return False
 
-def generate_testssl_report(name, ip_addr):
-    """Prepare the bash command and execute with $ip_addr as the target.
-        Creates a .json file in ./results/
+def generate_report(name, ip_addr):
+    """Prepare the flags/arguments for testssl and create a subprocess
+       calling the testssl.sh executable. Results are written
+       to an output file. Format `dir/www.example.com_20180307-164136.json`
+       Returns the path to the json file created.
     """
-    # set arguments for flags
+    # Prepare array of arguments to be passed to subprocess
+    args = []
+
+    # change this value if you move testssl for whatever reason
+    path_to_executable = './testssl.sh/testssl.sh'
+
+    # make sure we have no troubles writing to files
     log_dir = './results/'
     if not os.path.exists(log_dir):
         os.makedirs(log_dir)
-    # dir/www.example.com_20180307-164136.json
     log_path = log_dir + name + '_' + time.strftime("%Y%m%d-%H%M%S") + '.json'
-    severity = 'HIGH' # adjust this according to your level of paranoia
-    flags = ' '.join([
-            '--vulnerable',
-            '--severity ' + severity,
-            '--quiet',
-            '--nodns',
+
+    # adjust this according to your level of paranoia
+    # good values are HIGH and CRITICAL
+    severity = 'HIGH' 
+
+    # This must be a string because it's a command line argument
+    timeout_in_seconds = '20'
+
+    # subprocess expects a flat array; flags with arguments 
+    #       must be separated into their own elements
+    flags = [
+            '--vulnerable', # check for vulnerabilties
+            '--warnings', # testssl.sh will still warn you if there will be a "drastic impact"
```

Comment:
@kousu This should address #15 but please let me know if this is the case if you don't mind doing the testing.